### PR TITLE
feat: Move all third-party deps into git submodules

### PR DESCRIPTION
**Pull request by joeyparrish**
This also normalizes the structure of all submodule folders so that
there is a clear place to put configurations, outside the submodule
source, but limited in scope to that folder.

Issue #1047 (New build system)

### CODE REVIEW

**Comment by joeyparrish** [comment]
Needed some adjustment for various platforms once I had it up on GitHub, but it's building everywhere now.  PTAL!

File: packager/tools/generate_license_notice.py [hunk 1]
```diff
@@ -13,34 +13,18 @@
...
-                 'android-cmake'),
+    # Used for testing, not in the binary build.
+    os.path.join('third_party', 'json'),
+    os.path.join('third_party', 'googletest'),
+
+    # The Java side of icu, not used in Packager.
+    os.path.join('third_party', 'icu', 'source', 'icu4j'),
+
+    # Dependencies of dependencies, not actually used in Packager.
+    os.path.join('third_party', 'libpng', 'source', 'contrib'),
```
Comment (kqyang) [inline packager/tools/generate_license_notice.py:24]: nit: Move it after line 27 so the two icu are grouped together.

**Review by kqyang — APPROVED** [review]
lgtm with a nit

File: packager/tools/generate_license_notice.py (hunk 1 above)
Comment (joeyparrish) [reply packager/tools/generate_license_notice.py:24]: Done

**Pull request merged**